### PR TITLE
encoder: More unexpected EOF handling

### PR DESCRIPTION
**Pull request by navytux**
One more error handling patch continuing yesterday's story. Please see details inside.

Thanks beforehand,
Kirill

### CODE REVIEW

File: ogorek.go [hunk 1]
```diff
@@ -531,10 +535,7 @@ func (d *Decoder) loadUnicode() error {
 		if err != nil {
 			return err
 		}
-		_, err = buf.WriteRune(r)
-		if err != nil {
-			return err
-		}
+		buf.WriteRune(r)
```
Comment (kisielk) [inline ogorek.go]: Why is the check for error not necessary any more?

Comment (navytux) [inline ogorek.go]: This change was prepared as separate patch to exactly answer this questions:

https://github.com/kisielk/og-rek/pull/27/commits/d266ce2fe588b146b04dc4316403fbb599abc0e6

To duplicate it here: bytes.Buffer promise not to return error from Write*() ever:

https://golang.org/pkg/bytes/#Buffer.Write
https://golang.org/pkg/bytes/#Buffer.WriteRune
...

("... returning its length and an error, which is always nil but is included to match bufio.Writer's WriteRune. The buffer is grown as needed; if it becomes too large, WriteRune will panic with ErrTooLarge.")

It was already relied upon by ogórek code in other place. Please see details in the patch.

Thanks.

**Comment by kisielk** [comment]
Cool. Thanks again!

**Comment by navytux** [comment]
Thanks for applying.

**Pull request merged**